Runtime/GameLiftCoreApi.cs

```diff
@@ -3,51 +3,48 @@
 
 #if !UNITY_SERVER
 
-using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using AmazonGameLiftPlugin.Core;
 using AmazonGameLiftPlugin.Core.ApiGatewayManagement;
 using AmazonGameLiftPlugin.Core.ApiGatewayManagement.Models;
 using AmazonGameLiftPlugin.Core.Latency;
 using AmazonGameLiftPlugin.Core.Latency.Models;
 using AmazonGameLiftPlugin.Core.Shared;
 using AmazonGameLiftPlugin.Core.UserIdentityManagement;
 using AmazonGameLiftPlugin.Core.UserIdentityManagement.Models;
+using UnityEngine;
 
 namespace AmazonGameLift.Runtime
 {
     public class GameLiftCoreApi
     {
         private readonly GameLiftConfiguration _configuration;
-        private readonly bool _isLocalMode;
+        private readonly bool _isAnywhereMode;
 
-        public GameLiftCoreApi(GameLiftConfiguration configuration)
+        protected GameLiftCoreApi(GameLiftConfiguration configuration)
         {
             _configuration = configuration;
             _userIdentity = new UserIdentity(new AmazonCognitoIdentityWrapper(configuration.AwsRegion));
             _apiGateway = new ApiGateway(_userIdentity, new JwtTokenExpirationCheck(), new HttpClientWrapper());
-
-            if (_configuration.ApiGatewayEndpoint.StartsWith("https://localhost"))
-            {
-                throw new ArgumentException("Invalid configuration: https is not supported by GameLift Local.", nameof(configuration));
-            }
-
-            if (_configuration.ApiGatewayEndpoint.StartsWith("http://localhost"))
+            
+            var clientSettings = Resources.FindObjectsOfTypeAll<GameLiftClientSettings>()[0];
+            if (_configuration.IsGameLiftAnywhere)
             {
-                var gameLiftClientWrapper = new AmazonGameLiftClientWrapper(_configuration.ApiGatewayEndpoint);
-                _localGame = new LocalGameAdapter(gameLiftClientWrapper);
-                _isLocalMode = true;
+                var gameLiftClientWrapper = new AmazonGameLiftClientWrapper(clientSettings.ProfileName);
+                _anywhereGame = new AnywhereGameServerAdapter(gameLiftClientWrapper, clientSettings.FleetID, clientSettings.FleetLocation);
+                _isAnywhereMode = true;
```

Comment:
Is the logic to handle managed SDK5 fleets separate? Is that a follow up CR?

Reply:
This is used for GameLiftAnywhere Testing. It is set via the Scriptable object the user creates. Managed EC2 Fleets were done in a different way than GameLiftLocal